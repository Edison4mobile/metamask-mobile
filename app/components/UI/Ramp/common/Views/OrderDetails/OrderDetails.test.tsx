import React from 'react';
import { screen } from '@testing-library/react-native';
import renderWithProvider from '../../../../../../util/test/renderWithProvider';
import OrderDetails from './OrderDetails';
import initialBackgroundState from '../../../../../../util/test/initial-background-state.json';
import { FiatOrder } from '../../../../../../reducers/fiatOrders';
import {
  FIAT_ORDER_PROVIDERS,
  FIAT_ORDER_STATES,
} from '../../../../../../constants/on-ramp';

const mockNavigate = jest.fn();
const mockSetNavigationOptions = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualReactNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...actualReactNavigation,
    useNavigation: () => ({
      navigate: mockNavigate,
      setOptions: mockSetNavigationOptions,
    }),
  };
});

type DeepPartial<BaseType> = {
  [key in keyof BaseType]?: DeepPartial<BaseType[key]>;
};

const testOrders: DeepPartial<FiatOrder> = {
  id: 'test-order-1',
  account: '0x0',
  network: '1',
  cryptoAmount: '0.01231324',
  orderType: 'BUY',
  state: FIAT_ORDER_STATES.COMPLETED,
  createdAt: 1697242033399,
  provider: FIAT_ORDER_PROVIDERS.AGGREGATOR,
  cryptocurrency: 'ETH',
  amount: '34.23',
  currency: 'USD',
  data: {
    cryptoCurrency: {
      decimals: 18,
      name: 'Ethereum',
      symbol: 'ETH',
    },
    provider: {
      name: 'Test Provider',
    },
  },
};

jest.mock('../../../../../../util/navigation/navUtils', () => ({
  ...jest.requireActual('../../../../../../util/navigation/navUtils'),
  useParams: jest.fn(() => ({
    orderId: 'SOME_ID',
  })),
}));

function render(Component: React.ReactElement, orders = testOrders) {
  return renderWithProvider(Component, {
    state: {
      engine: {
        backgroundState: {
          ...initialBackgroundState,
          PreferencesController: {
            selectedAddress: '0x0',
            identities: {
              '0x0': {
                address: '0x0',
                name: 'Account 1',
              },
            },
          },
          NetworkController: {
            network: '1',
            providerConfig: {
              ticker: 'ETH',
              type: 'mainnet',
              chainId: '1',
            },
          },
        },
      },
      fiatOrders: {
        orders: [orders] as FiatOrder[],
      },
    },
  });
}
describe('OrderDetails', () => {
  it('renders an empty screen layout if there is no order', async () => {
    render(<OrderDetails />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
