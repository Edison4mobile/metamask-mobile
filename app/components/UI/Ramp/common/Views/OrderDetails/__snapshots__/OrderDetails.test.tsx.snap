// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetails renders an empty screen layout if there is no order 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        },
        undefined,
      ]
    }
  />
</RCTSafeAreaView>
`;
